neptune/instance.go

```diff
@@ -83,7 +83,7 @@ func (n *NeptuneDB) CreateInstance(ctx context.Context, instanceID string, csvHe
 		return nil
 	}
 
-	create := fmt.Sprintf(query.CreateInstance, instanceID, strings.Join(csvHeaders, ","))
+	create := fmt.Sprintf(query.CreateInstance, instanceID, instanceID, strings.Join(csvHeaders, ","))
```

Comment:
Would we be better off deliberately making this a _different_ unique value? Just thinking it feels like this is an internal ID specifically for database storage/search optimisation making it completely separate from the instance ID feels like a nice clean break and means you aren't tied to using the instance ID if we ever needed to change it in future.

Happy to be corrected on this as I have no detailed knowledge of this task 😄

Reply:
This is a good point, and would be the case in other databases. Though Neptune has no mechanism of creating indexes as you would with other databases. Instead of creating a primary key index as you would in relational databases, In Neptune you have to create a custom ID property value to get the performance of a primary key index lookup. At first we used a combination of label and property as a composite primary key, but this was one of the main causes of the query performance issues. This was the first thing highlighted by the Neptune expert when he looked at the query.

The ID property is stored just as a string, so custom ID's have no negative impact: https://docs.aws.amazon.com/neptune/latest/userguide/access-graph-gremlin-differences.html#w3aac12c20c10c15c25

Reply:
As discussed this sounds sensible. Im not that well informed on Neptune/Gremlin but figured it's better to ask questions anyway and let those in the know confirm it's the best thing to do.